src/intermediates/layer.jl

```diff
@@ -32,9 +32,36 @@ function layer(plot1::VegaVisualization, plot2::VegaVisualization)
     yloc2 = findfirst([z.name == "y" for z in overlay.scales])
 
     #Overwrite baseplot domain value
-    baseplot.scales[xloc1].domain = VegaDataRef(fields = [baseplot.scales[xloc1].domain, overlay.scales[xloc2].domain])
-    baseplot.scales[yloc1].domain = VegaDataRef(fields = [baseplot.scales[yloc1].domain, overlay.scales[yloc2].domain])
+    if baseplot.scales[xloc1].domain.fields == nothing
+        xdomain1=baseplot.scales[xloc1].domain
+    else
+        xdomain1=baseplot.scales[xloc1].domain.fields
+    end
+    if overlay.scales[xloc2].domain.fields == nothing
+        xdomain2=overlay.scales[xloc2].domain
+    else
+        xdomain2=overlay.scales[xloc2].domain.fields
+    end
+
+    if baseplot.scales[yloc1].domain.fields == nothing
+        ydomain1=baseplot.scales[yloc1].domain
+    else
+        ydomain1=baseplot.scales[yloc1].domain.fields
+    end
+    if overlay.scales[yloc2].domain.fields == nothing
+        ydomain2=overlay.scales[yloc2].domain
+    else
+        ydomain2=overlay.scales[yloc2].domain.fields
+    end
+
+    baseplot.scales[xloc1].domain = VegaDataRef(fields =
+        [xdomain1, xdomain2;])
+    baseplot.scales[yloc1].domain = VegaDataRef(fields =
+        [ydomain1, ydomain2;])
 
     return baseplot
 
-end
\ No newline at end of file
+end
+
+import Base.+
++(plot1::VegaVisualization, plot2::VegaVisualization)=layer(plot1,plot2)
```

Comment:
If you're suggesting adding this as a feature, could you put it into a separate PR for discussion? I'm generally against this type of syntax (1) because I don't like ggplot style syntax and 2) more theoretically I'm not sure this operation is really a commutative one (though, I don't know that it isn't either).